runtime/drivers/pinot/olap.go

```diff
@@ -0,0 +1,300 @@
+}
+
+func (i informationSchema) All(ctx context.Context) ([]*drivers.Table, error) {
+	// query /tables endpoint, for each table name, query /tables/{tableName}/schema
+	var tableNames []string
+	// Get table names
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, i.c.baseURL+"/tables", http.NoBody)
+	resp, err := i.c.metaClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	var respBody map[string][]string
+	err = json.NewDecoder(resp.Body).Decode(&respBody)
+	if err != nil {
+		return nil, err
+	}
+	tableNames = respBody["tables"]
+
+	var tables []*drivers.Table
+	for _, tableName := range tableNames {
+		table, err := i.Lookup(ctx, "", "", tableName)
+		if err != nil {
+			fmt.Printf("Error fetching schema for table %s: %v\n", tableName, err)
+			continue
+		}
+		tables = append(tables, table)
+	}
+
+	return tables, nil
+}
+
+func (i informationSchema) Lookup(ctx context.Context, db, schema, name string) (*drivers.Table, error) {
+	req, _ := http.NewRequestWithContext(ctx, http.MethodGet, i.c.baseURL+"/tables/"+name+"/schema", http.NoBody)
+	resp, err := i.c.metaClient.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	var schemaResponse pinotSchema
+	err = json.NewDecoder(resp.Body).Decode(&schemaResponse)
+	if err != nil {
+		return nil, err
+	}
+
+	unsupportedCols := make(map[string]string)
+	var schemaFields []*runtimev1.StructType_Field
+	for _, field := range schemaResponse.DimensionFieldSpecs {
+		singleValueField := true
+		if field.SingleValueField != nil {
+			singleValueField = *field.SingleValueField
+		}
+		if !singleValueField {
+			// Skip array fields for now
+			unsupportedCols[field.Name] = field.DataType + "_ARRAY"
+			continue
+		}
+		schemaFields = append(schemaFields, &runtimev1.StructType_Field{Name: field.Name, Type: databaseTypeToPB(field.DataType, !field.NotNull, singleValueField)})
+	}
+	for _, field := range schemaResponse.MetricFieldSpecs {
+		singleValueField := true
+		if field.SingleValueField != nil {
+			singleValueField = *field.SingleValueField
+		}
+		if !singleValueField {
+			// Skip array fields for now
+			unsupportedCols[field.Name] = field.DataType + "_ARRAY"
+			continue
+		}
+		schemaFields = append(schemaFields, &runtimev1.StructType_Field{Name: field.Name, Type: databaseTypeToPB(field.DataType, !field.NotNull, singleValueField)})
+	}
+	for _, field := range schemaResponse.DateTimeFieldSpecs {
+		if field.DataType != "TIMESTAMP" {
+			unsupportedCols[field.Name] = field.DataType + "_(DATE_TIME_FIELD)"
+			continue
+		}
+		schemaFields = append(schemaFields, &runtimev1.StructType_Field{Name: field.Name, Type: databaseTypeToPB(field.DataType, !field.NotNull, true)})
+	}
```

Comment:
Can we move date time fields up before dimension fields? Usually, people would expect this order: time dims, other dims, metric dims.